test/core/pipeline/test_utils.py

```diff
@@ -184,78 +185,89 @@ def test_queue_ordering_parametrized(empty_queue, items):
         assert empty_queue.pop() == (priority, item)
 
 
+class Copyable:
+    def __init__(self, name="copyable"):
+        self.name = name
+
+
+class NotCopyable:
+    def __init__(self, name="not_copyable"):
+        self.name = name
+
+    def __deepcopy__(self, memo):
+        raise TypeError("This object cannot be deepcopied.")
+
+
 class TestDeepcopyWithFallback:
     def test_deepcopy_with_fallback_copyable(self, caplog):
-        tool = Tool(
-            name="weather",
-            description="Get weather report",
-            parameters={"type": "object", "properties": {"city": {"type": "string"}}, "required": ["city"]},
-            function=get_weather_report,
-        )
-        original = {"tools": tool}
+        original = {"class": Copyable()}
         with caplog.at_level(logging.INFO):
             copy = deepcopy_with_fallback(original)
             assert "Deepcopy failed for object of type" not in caplog.text
-        assert copy["tools"] == original["tools"]
+        assert copy["class"].name == original["class"].name
```

Comment:
probably this and other tests could be simplified using the `is` keyword

`copy is original` should be `True` if the two variables refer to the same object

Reply:
thanks! simplified the tests